### ISSUE TITLE
chunking-loop needs a throttle

### ISSUE THREAD

**Issue by jti-lanl** [pftool/pftool#35]
The "chunking-loop" in process_stat_buffer() iterates generating chunks and sending them to the manager.  The manager receives these (as PROCESCMD) and pushes them onto a linked-list, where they are eventually doled out to workers that need something to do.

In some situations (e.g. gigantic file that will yield many chunks), the chunker can generate these TBD chunks faster than other workers can consume them, causing the manager to eat VM and grow indefinitely.

* One first-cut solution would be to have workers in the chunking-loop look at the size of the linked list.  If it's above some threshold (e.g. 1M entries), then sleep for e.g. 60 seconds and try again.

* For finer control, instead of polling, there could be a lock.  Worker sends message saying "unlock me when list-size falls below threshold/2", then waits on a per-worker lock.  Then master checks the list size whenever it decreases, and when it falls below the target it unlocks all waiting chunkers.

* A probably-unnecessarily-complicated approach would be for the chunking loop to bail when the list is full, so it could join the other workers doing the moving of chunks.  Then the manager could redeploy someone to resume chunking when the list got below e.g. threshold/2.  The problem here is that it implies worker_readdir() could return without having finished it's job.  Some state would be needed so the re-deploy could pick up where the previous guy left off.  If there are plenty of tasks, then this approach seems like unnecessary complication.

This last approach could be necessary if it's possible for there to be multiple large files such that all the workers are stuck in chunking loops, waiting for room in the linked list, with nobody doing moves.  We could avoid that by segregating workers (as was apparently the way things were done, once upon a time).

**Comment by garygrider** [comment]
Yes, in the past there were different procs for different tasks and the copy procs got dispatched at a priority to all other tasks and the mother watched the list size
And woudlnt displatch more lookup work until the list was smaller
This problem happened over a decade ago to the original version and dealt with it.
The readdir thing also had a – I detected I am in grass, I am going to punt and tell the mother that this is a grassy directory so hand it out in chunks via seekdir

I really miss the old code I guess but I am sure the software engineers would have really hated it

**Comment by thewacokid** [comment]
This was reworked in recent patches to throttle work queues if they amass enough work to avoid running out of memory.
